README.md

```diff
@@ -1,5 +1,234 @@
+| `develop`    | Ongoing development for future releases of the staging networks. This branch gets merged into `staging` on releases.           |
+| `production` | The latest releases for the IOTA network.                                                                                      |
+| `staging`    | The latest releases for the Shimmer network.                                                                                   |
+| other        | Other branches that may reflect current projects. Like `develop`, they will find their way into `staging` once they are ready. |
+
+## Before You Start
+
+This file is focused on Rust. Please refer to
+the [Python](bindings/python/README.md), [Node.js](bindings/nodejs/README.md) and [Wasm](bindings/wasm/README.md)
+instructions if you want information on installing and using them.
+
+## Requirements
+
+The IOTA SDK requires `Rust` and `Cargo`. You can find installation instructions in
+the [Rust documentation](https://doc.rust-lang.org/cargo/getting-started/installation.html).
+
+We recommend that you update the Rust compiler to the latest stable version first:
+
+```shell
+rustup update stable
+```
+
+
+### Dependencies
+
+You must also install `cmake`, `clang`, and `openssl`. You may need to install additional build tools on your system to
+run the build process successfully using Cargo.
+
+#### Windows
+
+You can download `cmake` from the [official website](https://cmake.org/download/). You can install `openssl`
+with [vcpkg](https://github.com/microsoft/vcpkg) or [chocolatey](https://chocolatey.org/).
+
+- Installing `openssl` with `vcpkg`:
+
+```
+./vcpkg.exe install openssl:x64-windows
+./vcpkg.exe integrate install
+# You may want to add this to the system environment variables since you'll need it to compile the crate
+set VCPKGRS_DYNAMIC=1
+```
+
+- Installing `openssl` with `chocolatey`:
+
+```
+choco install openssl
+# You may need to set the OPENSSL_DIR environment variable
+set OPENSSL_DIR="C:\Program Files\OpenSSL-Win64"
+```
+
+#### macOS
+
+You can install `cmake` and `openssl` with `Homebrew`:
+
+```
+brew install cmake openssl@1.1
+```
+
+#### Linux
+
+You can install `cmake`, `clang`, and `openssl` with your distro's package manager or download them from their websites.
+On Debian and Ubuntu, you will also need the `build-essential` and `libudev-dev` packages.
+
+## Getting Started
+
+### Install the IOTA SDK
+
+To start using the IOTA SDK in your Rust project, you can include the following dependencies in your `Cargo.toml` file:
+
+```toml
+[dependencies]
+iota-sdk = { branch = "develop" }
+```
+
+### Usage
+
+#### Client
+
+To use the Client module, you simply need to create a `Client`.
+
+```rust
+use iota_sdk::client::{
+    Client,
+};
+
+let client = Client::builder()
+    .with_node('https://api.testnet.shimmer.network')? // Insert your node URL here
+    .finish()
+    .await?;
+        
+let info = client.get_info().await?;
+println!("Node Info: {info:?}")
+
+Ok(())
```

Comment:
There is no main function and also not other function at all, so one can't just run it like this and the `Ok(())` is useless